### PR TITLE
Fix LuisAdapterRecognizer by correctly cloning the temporary TurnContext for adaptiing to legacy LuisRecognizer

### LINKED ISSUE #4536: LuisAdaptiveRecognizer incorrectly clones turncontext leading to middleware being ignored.
## Version

4.10.0

## Describe the bug

In Adaptive libraries we moved to a new Recognzer model which allows you to recognize arbitrary Activity objects.  The legacy LuisRecognizer implements IRecognzier which is TurnContext based and only recognizes the turnContext.Activity.  To prevent us from forking the old recognizer code, we simply created a cloned TurnContext with TurnContext.Activity=ActivityToRecgonize.

The issue is that we didn't do a full clone of the TurnContext, namely we dropped the internal middleware queues that the TurnContext has.  This means that outgoing messages from the Luis (aka trace messages) were not sent through middleware and so not logged into transcripts at all.  

## To Reproduce

1.  Add transcriptmiddleware
2.  call LuisRecognizer
3.  Trace will show up in emulator output, but not in transcript, because the transcript middleware is bypassed.

## Expected behavior

Middleware should be invoked by LuisRecognizer SendActivities.

### PR DESCRIPTION
**Pull request by tomlm**
Fixes #4536 

## Description
added TurnContext ctor which takes TurnContext to clone from and activity and which maintains all properties.

## Testing
added tests

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder/TurnContext.cs [hunk 1]
```diff
@@ -41,6 +41,37 @@ public TurnContext(BotAdapter adapter, Activity activity)
             Activity = activity ?? throw new ArgumentNullException(nameof(activity));
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TurnContext"/> class from another turncontext class.
```
Comment (gabog) [inline libraries/Microsoft.Bot.Builder/TurnContext.cs:45]: I would add something like "and sets the new instance activity property to the given activity" so the purpose of the constructor is more evident what it does from just reading the summary.

File: tests/Microsoft.Bot.Builder.Tests/TurnContextTests.cs [hunk 2]
```diff
@@ -35,6 +41,21 @@ public void Constructor()
...
+            var c1 = new TurnContext(new SimpleAdapter(), new Activity() { Text = "one" });
+            c1.TurnState.Add("x", "test");
+            c1.OnSendActivities((context, activities, next) => next());
+            c1.OnDeleteActivity((context, activity, next) => next());
+            c1.OnUpdateActivity((context, activity, next) => next());
+            var c2 = new TurnContext(c1, new Activity() { Text = "two" });
+            Assert.Equal("one", c1.Activity.Text);
+            Assert.Equal("two", c2.Activity.Text);
+            Assert.Equal(c1.Adapter, c2.Adapter);
+            Assert.Equal(c1.TurnState, c2.TurnState);
```
Comment (gabog) [inline tests/Microsoft.Bot.Builder.Tests/TurnContextTests.cs:56]: Can we assert what happens if `if (turnContext is TurnContext tc)` is true to make sure we cover that block too?

File: libraries/Microsoft.Bot.Builder.AI.LUIS/LuisAdaptiveRecognizer.cs [hunk 3]
```diff
@@ -40,6 +40,13 @@ public LuisAdaptiveRecognizer()
         [JsonProperty("applicationId")]
         public StringExpression ApplicationId { get; set; }
 
+        /// <summary>
+        /// Gets or sets LUIS version.
+        /// </summary>
+        /// <value>application version.</value>
+        [JsonProperty("version")]
+        public StringExpression Version { get; set; }
```
Comment (gabog) [inline libraries/Microsoft.Bot.Builder.AI.LUIS/LuisAdaptiveRecognizer.cs:48]: Why is this needed as part of this PR?

File: libraries/Microsoft.Bot.Builder.AI.LUIS/Schemas/Microsoft.LuisRecognizer.schema [hunk 4]
```diff
@@ -16,6 +16,11 @@
             "title": "LUIS Application ID",
             "description": "Application ID for your model from the LUIS service."
         },
+        "version": {
+            "$ref": "schema:#/definitions/stringExpression",
+            "title": "LUIS Version",
+            "description": "Optional version to target.  If null then predictionOptions.Slot is used."
+        },
         "endpoint": {
```
Comment (chrimc62) [inline libraries/Microsoft.Bot.Builder.AI.LUIS/Schemas/Microsoft.LuisRecognizer.schema:24]: I still don't like promoting this to the top given the presence in prediction options.

File: libraries/Microsoft.Bot.Builder.AI.LUIS/LuisAdaptiveRecognizer.cs (hunk 3 above)
Comment (chrimc62) [reply libraries/Microsoft.Bot.Builder.AI.LUIS/LuisAdaptiveRecognizer.cs:48]: And this is a bad idea--version is already in prediction options.

---
In reply to: [480166975](https://github.com/microsoft/botbuilder-dotnet/pull/4537#discussion_r480166975) [](ancestors = 480166975)

Comment (tomlm) [reply libraries/Microsoft.Bot.Builder.AI.LUIS/LuisAdaptiveRecognizer.cs:48]: not supposed to be, got my PRs wires crossed.


---
In reply to: [481307166](https://github.com/microsoft/botbuilder-dotnet/pull/4537#discussion_r481307166) [](ancestors = 481307166,480166975)

File: libraries/Microsoft.Bot.Builder.AI.LUIS/Schemas/Microsoft.LuisRecognizer.schema (hunk 4 above)
Comment (tomlm) [reply libraries/Microsoft.Bot.Builder.AI.LUIS/Schemas/Microsoft.LuisRecognizer.schema:24]: this was not supposed to be in this PR

---
In reply to: [481305906](https://github.com/microsoft/botbuilder-dotnet/pull/4537#discussion_r481305906) [](ancestors = 481305906)

**Review by chrimc62 — APPROVED** [review]
:shipit:

**Review by gabog — APPROVED** [review]
